Add basic unit tests with pytest and fix Python 3.9 compatibility

SWE-Flow opened the pull request:
Fixes issue #1

This PR adds a basic set of unit tests using pytest to help catch regressions in the codebase. The implementation includes:

- Setup of pytest configuration in pyproject.toml
- Unit tests for the core functionality in the hooks and trace modules
- Fixed Python 3.9 compatibility issues by replacing the Union pipe operator (`|`) with `Union[type, None]` syntax

These tests will help ensure code quality and catch regressions as the project evolves.

SWE-Flow committed:
Add basic unit tests with pytest

Hambaobao commented:
@SWE-Flow 

The unit-test failed on Python 3.9. Here is the log:
```bash
============================= test session starts ==============================
platform linux -- Python 3.9.22, pytest-8.4.0, pluggy-1.6.0
rootdir: /home/runner/work/SWE-Flow-Trace/SWE-Flow-Trace
configfile: pyproject.toml
testpaths: tests
plugins: mock-3.14.1
collected 0 items / 2 errors

==================================== ERRORS ====================================
_____________________ ERROR collecting tests/test_hooks.py _____________________
tests/test_hooks.py:8: in <module>
    from sweflow_trace.python.hooks import CallTracer
sweflow_trace/python/hooks.py:13: in <module>
    class CallTracer:
sweflow_trace/python/hooks.py:15: in CallTracer
    def __init__(self, base_dir: str | None = None):
E   TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
_____________________ ERROR collecting tests/test_trace.py _____________________
tests/test_trace.py:8: in <module>
    from sweflow_trace.python.trace import (
sweflow_trace/python/trace.py:84: in <module>
    max_tests: int | None = None,
E   TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
=========================== short test summary info ============================
ERROR tests/test_hooks.py - TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
ERROR tests/test_trace.py - TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
!!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!
============================== 2 errors in 0.19s ===============================
```

By the way, would you remove the `run_tests.sh` file, as we need to keep the project clean.

SWE-Flow committed:
Add basic unit tests with pytest and fix Python 3.9 compatibility

SWE-Flow commented:
I've fixed the Python 3.9 compatibility issues by replacing the Union pipe operator (`|`) with the `typing.Union` syntax. The pipe operator for Union types was introduced in Python 3.10, so we need to use the older syntax for Python 3.9 compatibility.

Changes made:
1. Replaced `str | None` with `Union[str, None]` or `Optional[str]`
2. Replaced `int | None` with `Union[int, None]` or `Optional[int]`
3. Added proper imports from the typing module

I've also removed the `run_tests.sh` file as requested to keep the project clean. All tests can now be run directly with the `pytest` command.
